test/unit/control-connection-tests.js

```diff
@@ -58,6 +58,21 @@ describe('ControlConnection', function () {
         done();
       });
     });
+    it('should resolve IPv4 and IPv6 addresses with non default port', function (done) {
+      if (!useLocalhost) {
+        return done();
+      }
+      var cc = new ControlConnection(clientOptions.extend({ contactPoints: ['localhost:9999'] }));
+      cc.getConnection = helper.callbackNoop;
+      cc.refreshOnConnection = helper.callbackNoop;
+      cc.init(function (err) {
+        assert.ifError(err);
+        var hosts = cc.hosts.values();
+        assert.strictEqual(hosts.length, 2);
```

Comment:
I think asserting that it resolves at least to ipv4 loopback address (127.0.0.1) is good enough. Otherwise it may fail (ie: travis).